### PR TITLE
ui: fix market view bugs

### PR DESCRIPTION
**Pull request by buck54321**
Resolves https://github.com/decred/dcrdex/issues/1886

### CODE REVIEW

File: client/webserver/site/src/js/app.ts [hunk 1]
```diff
@@ -525,17 +513,18 @@ export default class Application {
...
-    // Handle type-specific updates.
-    this.log('notes', 'notify', note)
+  updateUser (note: CoreNote) {
+    const { user, assets, walletMap } = this
+    if (note.type === 'fiatrateupdate') {
+      this.fiatRatesMap = (note as RateNote).fiatRates
+      return
+    }
+    // Some notes can be received before we get a User during login.
+    if (!user) return
```
[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/app.ts:523]: I see these errors on master.

[the remark this answers is not in the document]
Comment (chappjc) [reply client/webserver/site/src/js/app.ts:523]: You mean the release branch or something in master before the restyle commit?
These js errors were almost completely squashed before, so I'm wondering why they're consistent now.

**Comment by buck54321** [comment]
> They look the same though it would be good if they had a way to tell them apart imo.

I'm still thinking about how best to do this.

File: client/webserver/site/src/js/app.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/app.ts:523]: I see it on the commit before the restyle.

[the remark this answers is not in the document]
Comment (chappjc) [reply client/webserver/site/src/js/app.ts:523]: I guess we were ignoring them before

[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/js/app.ts:523]: Must've been. You have to force a shutdown to see them and they're innocuous. There would be no indication anything went wrong without the console error. I had some awareness that things were racy there, but I hadn't gotten into it.

**Comment by JoeGruffins** [comment]
Do the markets view coin id hyperlinks still work? I can't click on dcr_btc testnet. I thought they worked before.

**Comment by chappjc** [comment]
I'm seeing this now when I load to initialize:

image

**Comment by chappjc** [comment]
First time seeing this, it doesn't ever go to synced:

image

If I F5 and reselect the asset, it goes to ready to pay.

**Comment by chappjc** [comment]
Horizontal search text box alignment looks fixed, but it seems offset a little high in the box, plus the magnifying glass icon is still shown when there is text (if you click away).

image

File: client/webserver/site/src/html/markets.tmpl [hunk 2]
```diff
@@ -410,15 +418,17 @@
                 </div>
               </div>{{- /* END USER ORDERS */ -}}
 
+              <a href="/orders" class="flex-center my-2 plainlink">[[[view order history]]]</a>
+
               {{- /* RECENT MATCHES */ -}}
-              <div id=recentMatchesBox  class="d-flex flex-column align-items-stretch px-3 pb-4">
+              <div id=recentMatchesBox  class="d-flex flex-column align-items-stretch px-3 pb-4 mt-2">
                 <div id="userOrdersHeader" class="text-center fs20 sans-light my-1">[[[Recent Matches]]]</div>
```
Comment (chappjc) [inline client/webserver/site/src/html/markets.tmpl:425]: Maybe a `brdrtop` and `my-3` on these two divs, respectively:

```diff
diff --git a/client/webserver/site/src/html/markets.tmpl b/client/webserver/site/src/html/markets.tmpl
index e3d7dc27..04e6c13f 100644
--- a/client/webserver/site/src/html/markets.tmpl
+++ b/client/webserver/site/src/html/markets.tmpl
@@ -421,8 +421,8 @@
               <a href="/orders" class="flex-center my-2 plainlink">[[[view order history]]]</a>
 
               {{- /* RECENT MATCHES */ -}}
-              <div id=recentMatchesBox  class="d-flex flex-column align-items-stretch px-3 pb-4 mt-2">
-                <div id="userOrdersHeader" class="text-center fs20 sans-light my-1">[[[Recent Matches]]]</div>
+              <div id=recentMatchesBox  class="d-flex flex-column align-items-stretch px-3 pb-4 mt-2 brdrtop">
+                <div id="userOrdersHeader" class="text-center fs20 sans-light my-3">[[[Recent Matches]]]</div>
                 <table id="recentMatchesTable" class="ordertable">
                   <thead>
                     <tr>
```

gives:

image

**Comment by chappjc** [comment]
Here's an illustration of what @martonp pointed out in https://github.com/decred/dcrdex/issues/1886#issuecomment-1272523234:

https://user-images.githubusercontent.com/9373513/195173060-ed5c1a68-4fab-4223-9ac1-1f2e3b4bf746.mp4

**Comment by buck54321** [comment]
Sidebar can't have a fixed width. I can increase the min width.

**Comment by buck54321** [comment]
> Horizontal search text box alignment looks fixed, but it seems offset a little high in the box, plus the magnifying glass icon is still shown when there is text (if you click away).
> 
> image

Is the off-focus 
magnifying glass a big deal? There's no way with CSS to tell if an input has text in it, so we would have to use a javascript hack.

**Comment by chappjc** [comment]
> There's no way with CSS to tell if an input has text in it, so we would have to use a javascript hack.

Oh.  I guess it's fine then.

**Comment by chappjc** [comment]
This seems to work for the ico-search thing:

```diff
--- a/client/webserver/site/src/css/market.scss
+++ b/client/webserver/site/src/css/market.scss
@@ -313,6 +313,10 @@ div[data-handler=markets] {
         opacity: 0.5;
       }
 
+      &:not(:placeholder-shown) + .ico-search {
+        display: none;
+      }
+
       &:focus + .ico-search {
         display: none;
       }
diff --git a/client/webserver/site/src/html/markets.tmpl b/client/webserver/site/src/html/markets.tmpl
index e3d7dc27..f66cfedf 100644
--- a/client/webserver/site/src/html/markets.tmpl
+++ b/client/webserver/site/src/html/markets.tmpl
@@ -17,7 +17,7 @@
 
       <div class="d-flex align-items-stretch" id="searchBoxV1">
         <div class="flex-grow-1 p-1 position-relative">
-          <input type="text" id="marketSearchV1" spellcheck="false">
+          <input type="text" id="marketSearchV1" placeholder=" " spellcheck="false">
           <div class="ico-search fs22"></div>
         </div>
         <div id="leftHider" class="flex-center pointer hoverbg">
@@ -421,8 +421,8 @@
               <a href="/orders" class="flex-center my-2 plainlink">[[[view order history]]]</a>
 
               {{- /* RECENT MATCHES */ -}}
-              <div id=recentMatchesBox  class="d-flex flex-column align-items-stretch px-3 pb-4 mt-2">
-                <div id="userOrdersHeader" class="text-center fs20 sans-light my-1">[[[Recent Matches]]]</div>
+              <div id=recentMatchesBox  class="d-flex flex-column align-items-stretch px-3 pb-4 mt-2 brdrtop">
+                <div id="userOrdersHeader" class="text-center fs20 sans-light my-3">[[[Recent Matches]]]</div>
                 <table id="recentMatchesTable" class="ordertable">
                   <thead>
                     <tr>
```

**Review by chappjc — APPROVED** [review]

**Pull request merged**